src/rank_llm/retrieve/service_retriever.py

```diff
@@ -0,0 +1,120 @@
+import json
+import requests
+from urllib import parse
+from enum import Enum
+from typing import Any, Dict, List, Union
+
+from rank_llm.data import Request, Candidate, Query
+from rank_llm.retrieve.pyserini_retriever import PyseriniRetriever, RetrievalMethod
+from rank_llm.retrieve.repo_info import HITS_INFO
+from rank_llm.retrieve.utils import compute_md5, download_cached_hits
+
+
+class RetrievalMode(Enum):
+    DATASET = "dataset"
+    CUSTOM = "custom"
+
+    def __str__(self):
+        return self.value
+
+
+class ServiceRetriever:
+    def __init__(
+        self,
+        dataset: Union[str, List[str], List[Dict[str, Any]]],
+        host: str,
+        retrieval_mode: RetrievalMode = RetrievalMode.DATASET,
+        retrieval_method: RetrievalMethod = RetrievalMethod.BM25,
```

Comment:
This should be part of the url params for the anserini get request similar to dataset.
If anserini rest api does not expose retrievers other than bm25, please leave a todo and raise an error similar to retrieval method

Reply:
If the class has no member variables, then should we still need a ServiceRetriever class?

Reply:
For now since `retrieval_method` and `retrieval_mode` have fixed values, the `retrieve` function doesn't need to know about them but eventually they need to become member variables for `retrieve` to access them.

Reply:
Sure